presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,4 +728,277 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                .setTableType(VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(viewProperties)
+                .withStorage(storage -> storage.setStorageFormat(VIEW_STORAGE_FORMAT))
+                .withStorage(storage -> storage.setLocation(""))
+                .setViewOriginalText(Optional.of(encodeMaterializedViewData(definition)))
+                .setViewExpandedText(Optional.of("/* Presto Materialized View */"));
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+        if (existing.isPresent() && replace) {
+            // drop the current storage table
+            String oldStorageTable = existing.get().getParameters().get(STORAGE_TABLE);
+            if (oldStorageTable != null) {
+                metastore.dropTable(identity, viewName.getSchemaName(), oldStorageTable, true);
+            }
+            // Replace the existing view definition
+            metastore.replaceTable(identity, viewName.getSchemaName(), viewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+        // create the view definition
+        metastore.createTable(identity, table, principalPrivileges);
+    }
+
+    @Override
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        final HiveIdentity identity = new HiveIdentity(session);
+        Table view = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName())
+                .orElseThrow(() -> new MaterializedViewNotFoundException(viewName));
+
+        String storageTableName = view.getParameters().get(STORAGE_TABLE);
+        if (storageTableName != null) {
+            try {
+                metastore.dropTable(identity, viewName.getSchemaName(), storageTableName, true);
+            }
+            catch (PrestoException e) {
+                log.warn(e, "Failed to drop storage table '%s' for materialized view '%s'", storageTableName, viewName);
+            }
+        }
+        metastore.dropTable(identity, viewName.getSchemaName(), viewName.getTableName(), true);
+    }
+
+    @Override
+    public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        transaction = icebergTable.newTransaction();
+
+        // delete before insert .. simulating overwrite
+        executeDelete(session, tableHandle);
```

Comment:
Good catch! 
I was surprised why it did not come up in my testing before. Explaining the refresh command would delete the data and we should see problems when we try to use/refresh the materialized view.
Turns out, when materialized view is fresh, early on, during analysis we determine that it is fresh and refresh can be skipped, so beginRefreshMaterializedView is not called and storage table data stays intact. When it is stale, explain would delete the data, but no harm done, since materialized view's storage table can't be used for querying anyway and any subsequent refresh would just delete the non-existent data. 
Given the above and the fact that we only have full refresh so far, this problem was not seen. 
I have now moved the delete to 'finishRefresh...', I need to send in the ConnectorTableHandle for the storage table to 'finishRefresh ..', I am adding those changes in the same PR in a separate commit.
@martint FYI for connector API changes and because this is fascinating.